commit 8667a004d6148351a5d66f67889291b8e7466941
Author: Justin Stitt <justinstitt@google.com>
Date:   Wed Mar 20 22:48:50 2024 +0000

    fbdev: fsl-diu-fb: replace deprecated strncpy with strscpy_pad
    
    strncpy() is deprecated for use on NUL-terminated destination strings
    [1] and as such we should prefer more robust and less ambiguous string
    interfaces.
    
    A better alternative is strscpy() as it guarantees NUL-termination on
    the destination buffer.
    
    Since we are eventually copying over to userspace, let's ensure we
    NUL-pad the destination buffer by using the pad variant of strscpy.
    - core/fb_chrdev.c:
    234 |   err = copy_to_user(&fix32->id, &fix->id, sizeof(fix32->id));
    
    Furthermore, we can use the new 2-argument variants of strscpy() and
    strscpy_pad() introduced by Commit e6584c3964f2f ("string: Allow
    2-argument strscpy()") to simplify the syntax even more.
    
    Link: https://www.kernel.org/doc/html/latest/process/deprecated.html#strncpy-on-nul-terminated-strings [1]
    Link: https://manpages.debian.org/testing/linux-manual-4.8/strscpy.9.en.html [2]
    Link: https://github.com/KSPP/linux/issues/90
    Cc: linux-hardening@vger.kernel.org
    Signed-off-by: Justin Stitt <justinstitt@google.com>
    Reviewed-by: Kees Cook <keescook@chromium.org>
    Signed-off-by: Helge Deller <deller@gmx.de>

diff --git a/drivers/video/fbdev/fsl-diu-fb.c b/drivers/video/fbdev/fsl-diu-fb.c
index 0191141657fd..ea37a60da10c 100644
--- a/drivers/video/fbdev/fsl-diu-fb.c
+++ b/drivers/video/fbdev/fsl-diu-fb.c
@@ -787,7 +787,7 @@ static void set_fix(struct fb_info *info)
 	struct fb_var_screeninfo *var = &info->var;
 	struct mfb_info *mfbi = info->par;
 
-	strncpy(fix->id, mfbi->id, sizeof(fix->id));
+	strscpy_pad(fix->id, mfbi->id);
 	fix->line_length = var->xres_virtual * var->bits_per_pixel / 8;
 	fix->type = FB_TYPE_PACKED_PIXELS;
 	fix->accel = FB_ACCEL_NONE;